python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1250 @@
+
+        # Deferred import of TARTS to handle cases where it's not in the build
+        try:
+            from tarts import NeuralActiveOpticsSys
+        except ImportError as e:
+            raise ImportError(
+                "TARTS neural network system not available. "
+                "Please ensure TARTS is installed and available in the Python path. "
+                f"Original error: {e}"
+            ) from e
+
+        # TARTS system handles None paths by creating new models with random
+        # weights
+        self.tarts = NeuralActiveOpticsSys(
+            os.path.expandvars(self.config.datasetParamPath),
+            os.path.expandvars(self.config.wavenetPath),
+            os.path.expandvars(self.config.alignetPath),
+            os.path.expandvars(self.config.aggregatornetPath),
+        )
+        self.tarts = self.tarts.eval()
+        # Use configurable crop size instead of hard-coded TARTS value
+        self.cropSize = self.config.cropSize
+        self.device = self.config.device
+
+        self.log.info("Initialized TARTS with dataset params: %s", self.config.datasetParamPath)
+        self.log.debug(
+            "Model paths - wavenet: %s, alignet: %s, aggregatornet: %s",
+            self.config.wavenetPath,
+            self.config.alignetPath,
+            self.config.aggregatornetPath,
+        )
+        self.log.info("Running on device: %s", self.device)
+        self.log.info(
+            "Defocal distances - intra: %.1f mm, extra: %.1f mm",
+            self.config.intraDfcDist,
+            self.config.extraDfcDist,
+        )
+
+        if self.device == "cpu":
+            self.tarts = self.tarts.cpu()
+
+            # Update device attributes for all sub-models to CPU
+            self.tarts.device_val = torch.device("cpu")
+            for model in [
+                self.tarts.alignnet_model,
+                self.tarts.wavenet_model,
+                self.tarts.aggregatornet_model,
+            ]:
+                if model is not None:
+                    model.device_val = torch.device("cpu")
+
+                    # Update device attributes for underlying models and CNNs
+                    if hasattr(model, "alignnet"):
+                        model.alignnet.device_val = torch.device("cpu")
+                        model.alignnet.cnn = model.alignnet.cnn.cpu()
+                    if hasattr(model, "wavenet"):
+                        model.wavenet.device_val = torch.device("cpu")
+                        model.wavenet.cnn = model.wavenet.cnn.cpu()
+            self.log.debug("Moved all sub-models and CNNs to CPU")
+        else:
+            self.tarts.to("cuda")
+            self.log.debug("Moved models to CUDA")
+
+        self.log.debug("TARTS crop size: %s", self.cropSize)
+
+    def validate(self) -> None:
+        """Validate configuration parameters."""
+        # Validate defocal distances
+        if abs(self.config.intraDfcDist) < self.MIN_DEFOCAL_DISTANCE_MM:
```

Comment:
Sorry where are you getting the 0.1 mm (100 um) value from? Like how did you determine that limit?